PowerEditor/src/WinControls/OpenSaveFileDialog/CustomFileDialog.cpp

```diff
@@ -362,31 +369,71 @@ class FileDialogEventHandler : public IFileDialogEvents, public IFileDialogContr
 	FileDialogEventHandler(FileDialogEventHandler&&) = delete;
 	FileDialogEventHandler& operator=(FileDialogEventHandler&&) = delete;
 
-	// Overrides window procedures for file name edit and ok button.
+	// Find window handles for file name edit and ok button.
 	// Call this as late as possible to ensure all the controls of the dialog are created.
-	void initControls()
+	bool findControls()
 	{
 		assert(_dialog);
-		com_ptr<IOleWindow> pOleWnd = _dialog;
-		if (pOleWnd)
+		if (HWND hwndDlg = getDialogHandle(_dialog))
```

Comment:
assignment in if clause should be typically avoided to find easily the wrong usage of = instead of ==, see https://wiki.sei.cmu.edu/confluence/display/c/EXP45-C.+Do+not+perform+assignments+in+selection+statements

Reply:
Right. However, on the other hand, it reduces the scope of a variable and makes code more compact.
But I agree, let's change it.